### ISSUE TITLE
Add parser support for hash style comments (#) in MariaDB, MySQL

### ISSUE THREAD

**Issue by Kidlike** [jOOQ/jOOQ#11333]
### Expected behavior
hash style comments to be parsed/ignored (https://mariadb.com/kb/en/comment-syntax/)

### Actual behavior
throws ParserException

### Steps to reproduce the problem
1. sample code:
```
DSL.using(SQLDialect.MARIADB).parser().parse("# this is a comment")
```
2. throws exception:
```
org.jooq.impl.ParserException: Unsupported query type: [1:1] [*]# this is a comment
	at org.jooq.impl.ParserContext.exception(ParserImpl.java:12258)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:1034)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:618)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:605)
```

### Versions

- jOOQ: 3.14.6
- Java:
```
openjdk 11.0.8 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.21.0, JRE 11 Linux amd64-64-Bit Compressed References 20200715_697 (JIT enabled, AOT enabled)
OpenJ9   - 34cf4c075
OMR      - 113e54219
JCL      - 95bb504fbb based on jdk-11.0.8+10)
```
- Database (include vendor): MariaDB

**Comment by lukaseder** [comment]
Thanks a lot for your feature request. We're doing that already for plain SQL templates (see #4182), but not in the parser yet. Makes perfect sense. However, you'll have to specify the `Settings.parseDialect` once this is implemented, because other RDBMS support the `#` symbol in identifiers.

**Comment by Kidlike** [comment]
Thanks for the fast response and action!

I know that using the SQL parser is not the main goal of this library, but I had this need in one of my projects, and surprisingly there's not much out there for Java...
I only found jOOQ and another one, but in the end neither could parse properly all our db scripts. Looks like MariaDB added too many custom commands :)

I ended up using a very basic parser that is included in Liquibase, which only gives you back a list of String tokens (random example: `["SET", "FOREIGN_KEY_CHECKS", "=", "0", ";"]`), and then handled the rest of my requirements myself with a small state machine that covers my needs.

**Comment by lukaseder** [comment]
The SQL parser is heavily used already today, in the code generator, throughout jOOQ's internals, in our integration tests, and increasingly also by customers who wish to migrate between dialects (see https://www.jooq.org/translate).

It's much more work than the jOOQ API, of course, because the API can support a subset of SQL and people are still fine. The parser has to support all of it to be usable. Or at least all of what *you're* using :)

We'll get there. There was no illusion of reaching a reasonable feature completeness in a short time, but we're already quite far. Currently, support for procedures, functions, and triggers are being added in jOOQ 3.15.

Your bug reports are greatly appreciated! If you have more bugs resulting from your tests, I'd love to hear about it! Many missing syntaxes can be supported in no time. And if something isn't really required to be understood by jOOQ, we have a special comment syntax which you may add:
https://www.jooq.org/doc/latest/manual/sql-building/dsl-context/custom-settings/settings-parser/

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0. Thanks again for the report!
